pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -890,3 +890,308 @@ func TestVolumeBinding(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSchedulableAfterCSIStorageCapacityChange(t *testing.T) {
```

Comment:
I'm not sure if a `type conversion error` test is really necessary, but can't the test be simplified this way?

```diff
diff --git a/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go b/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go
index 040dd8a8683..cebb6aec9ce 100644
--- a/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go
+++ b/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go
@@ -892,16 +892,15 @@ func TestVolumeBinding(t *testing.T) {
 }

 func TestIsSchedulableAfterCSIStorageCapacityChange(t *testing.T) {
-       table := []struct {
-               name           string
-               pod            *v1.Pod
-               sc             *storagev1.StorageClass
-               oldCap         *storagev1.CSIStorageCapacity
-               newCap         *storagev1.CSIStorageCapacity
-               pvcLister      tf.PersistentVolumeClaimLister
-               useEmptyStruct bool
-               err            bool
-               expect         framework.QueueingHint
+       tests := []struct {
+               name      string
+               pod       *v1.Pod
+               sc        *storagev1.StorageClass
+               oldCap    interface{}
+               newCap    interface{}
+               pvcLister tf.PersistentVolumeClaimLister
+               err       bool
+               expect    framework.QueueingHint
        }{
                {
                        name: "pod has no pvcs",
@@ -1100,10 +1099,11 @@ func TestIsSchedulableAfterCSIStorageCapacityChange(t *testing.T) {
                        expect: framework.Queue,
                },
                {
-                       name:           "type conversion error",
-                       useEmptyStruct: true,
-                       err:            true,
-                       expect:         framework.Queue,
+                       name:   "type conversion error",
+                       oldCap: new(struct{}),
+                       newCap: new(struct{}),
+                       err:    true,
+                       expect: framework.Queue,
                },
                {
                        name: "pod has pvcs but pvc not found",
@@ -1173,24 +1173,17 @@ func TestIsSchedulableAfterCSIStorageCapacityChange(t *testing.T) {
                },
        }

-       for _, item := range table {
-               t.Run(item.name, func(t *testing.T) {
-                       pl := &VolumeBinding{PVCLister: item.pvcLister}
+       for _, test := range tests {
+               t.Run(test.name, func(t *testing.T) {
+                       pl := &VolumeBinding{PVCLister: test.pvcLister}
                        logger, _ := ktesting.NewTestContext(t)

-                       var qhint framework.QueueingHint
-                       var err error
-                       if item.useEmptyStruct {
-                               qhint, err = pl.isSchedulableAfterCSIStorageCapacityChange(logger, item.pod, new(struct{}), new(struct{}))
-                       } else {
-                               qhint, err = pl.isSchedulableAfterCSIStorageCapacityChange(logger, item.pod, item.oldCap, item.newCap)
-                       }
-
-                       if (item.err && err == nil) || (!item.err && err != nil) {
-                               t.Errorf("isSchedulableAfterCSIStorageCapacityChange failed - got: %q", err)
+                       qhint, err := pl.isSchedulableAfterCSIStorageCapacityChange(logger, test.pod, test.oldCap, test.newCap)
+                       if (err != nil) != test.err {
+                               t.Errorf("isSchedulableAfterCSIStorageCapacityChange() error = %v, want %v", err, test.err)
                        }
-                       if qhint != item.expect {
-                               t.Errorf("QHint does not match: %v, want: %v", qhint, item.expect)
+                       if qhint != test.expect {
+                               t.Errorf("QHint does not match: %v, want: %v", qhint, test.expect)
                        }
                })
        }
```

Reply:
This test is necessary for the error checking of this `util.As` function.

https://github.com/kubernetes/kubernetes/blob/411d7ae7c02f069c19298028401d4114eacdef97/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L130-L133

Reply:
I fiex it.

Reply:
> This test is necessary for the error checking of this util.As function.

Yes, I understand that. I think it's to increase test coverage, but my point is whether it's really necessary to test something obvious. Looking at the implementation of other QHints, it doesn't seem like they are being verified with unit tests.